### PR TITLE
[eas-cli] mark `default` image tag and old images as deprecated and add new image tags to `eas.schema.json`

### PR DESCRIPTION
**Pull request by szdziedzic**
# Why

The current `default` image tag is confusing. It's not even the true default. Changing it breaks people's builds and so on...

Let's deprecate it.

Add a new `auto` tag that is an explicit way of opting into the true default behavior (auto-resolving image when it is not specified in `eas.json`).

Add `sdk-XY` tags

Deprecate images with Xcode < 15 because of https://expo.dev/changelog/2024/03-29-eas-build-upcoming-ios-images-updates. Advise people on how to switch to Xcode 15 image.

Deprecate old Android images.

More context: https://exponent-internal.slack.com/archives/C9PRD479V/p1710257552292179

https://github.com/expo/expo/pull/27852
https://github.com/expo/turtle-v2/pull/1736

# How

Mark the `default` image tag as deprecated

Add a new `auto` tag that is an explicit way of opting into the true default behavior (auto-resolving image when it is not specified in `eas.json`).

Add `sdk-XY` tags

Deprecate images with Xcode < 15 because of https://expo.dev/changelog/2024/03-29-eas-build-upcoming-ios-images-updates. Advise people on how to switch to Xcode 15 image.

Deprecate old Android images with JDK 8 (used in auto image resolving logic for RN < 0.68.0) and NDK r19c (used for SDK < 46).

# Test Plan

Test manually

Screenshot 2024-03-29 at 13 55 55
Screenshot 2024-03-29 at 13 55 29

### CODE REVIEW

**Comment by szdziedzic** [comment]
/changelog-entry chore Deprecate the `default` image tag.

**Comment by szdziedzic** [comment]
/changelog-entry new-feature Add `auto`, `sdk-50` and `sdk-49` image tags

**Comment by szdziedzic** [comment]
/changelog-entry chore Deprecate Android images with JDK 8 and NDK `r19c`

**Review by khamilowicz — APPROVED** [review]

File: packages/eas-json/schema/eas.schema.json [hunk 1]
```diff
@@ -398,33 +377,46 @@
...
+                "macos-monterey-12.6-xcode-14.1",
+                "macos-monterey-12.6-xcode-14.0"
               ],
               "markdownEnumDescriptions": [
-                "This image is deprecated, please use one of the more recent images instead.",
-                "This image is deprecated, please use one of the more recent images instead.",
-                "This image is deprecated, please use one of the more recent images instead."
+                "This image tag is deprecated, please use other image tags or images directly instead.",
+                "This image tag is deprecated, please use other image tags or images directly instead.",
+                "This image is using Xcode in version lower then 15. Apple's new requirement states that starting on April 29th, 2024, apps uploaded to App Store Connect must be built with Xcode 15 for iOS 17, iPadOS 17, tvOS 17, or watchOS 10. Start using Xcode 15 image today by setting `macos-ventura-13.6-xcode-15.2` or `latest` as your image value. This image will be removed from EAS Build on May 29th, 2024. Learn more: expo.dev/changelog/2024/03-29-eas-build-upcoming-ios-images-updates.",
```
Comment (sjchmiela) [inline packages/eas-json/schema/eas.schema.json:412]: Should the URL have https? I _think_ I saw vscode linkifying links in descriptions?

**Review by sjchmiela — APPROVED** [review]
I'm wondering if we should also include some simple suggestion to follow for developers in the eas-cli Android warning. Not knowing what I know about images I wouldn't want to need to pick a replacement for a deprecated Android image. Which? Why?

**Comment by szdziedzic** [comment]
@sjchmiela You are right, I will remove the Android image warnings for now because I will be able to give people more helpful instructions on what to do next once https://github.com/expo/turtle-v2/pull/1744 lands and we have more images with newer NDKs better suited for SDK 49 and 50. Will open a separate PR to add Android images warnings 👍.

**Pull request merged**